log.go

```diff
@@ -150,66 +166,106 @@ func Level(level int) {
 	std.Level(level)
 }
 
-func (log *Log) SetOutput(writer io.Writer) {
-	internal_logger.SetOutput(writer)
+// thread-safe, use for accessing level in exported methods
+func (log *Log) getLevel() int {
+	log.mu.RLock()
+	defer log.mu.RUnlock()
+
+	return log.level
 }
 
-func SetOutput(path string) (err error) {
+func (log *Log) SetOutput(path string) (err error) {
+	log.mu.Lock()
+	defer log.mu.Unlock()
+
+	return log.setOutput(path)
+}
+
+// Not thread-safe
+func (log *Log) setOutput(path string) (err error) {
 	file, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0600)
 	if err != nil {
 		err = fmt.Errorf("Unable to open log file - err='%s'", err.Error())
 		return
 	}
 
-	std.logFileName = path
-	std.logWriter = file
-
-	std.SetOutput(file)
-
+	log.logFileName = path
+	log.logWriter = file
+	internal_logger.SetOutput(file)
 	return
 }
 
-func CloseOutput() (err error) {
-	err = std.logWriter.Close()
+func SetOutput(path string) (err error) {
+	return std.SetOutput(path)
+}
 
-	return
+func (log *Log) CloseOutput() (err error) {
+	log.mu.Lock()
+	defer log.mu.Unlock()
+
+	return log.closeOutput()
 }
 
-func rotateLog() {
+// Not thread-safe
+func (log *Log) closeOutput() (err error) {
+	return log.logWriter.Close()
+}
 
+func CloseOutput() (err error) {
```

Comment:
Is this used anywhere? Seems like people will be using `log.CloseOutput()` only?